### PR TITLE
Allow that we run php-cs-fixer on PHP8

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: php-cs-fixer [hunk 1]
```diff
@@ -23,8 +23,8 @@ if (defined('HHVM_VERSION_ID')) {
     } else {
         exit(1);
     }
-} elseif (!defined('PHP_VERSION_ID') || \PHP_VERSION_ID < 50600 || \PHP_VERSION_ID >= 70500) {
-    fwrite(STDERR, "PHP needs to be a minimum version of PHP 5.6.0 and maximum version of PHP 7.4.*.\n");
+} elseif (!defined('PHP_VERSION_ID') || \PHP_VERSION_ID < 50600 || \PHP_VERSION_ID >= 80100) {
+    fwrite(STDERR, "PHP needs to be a minimum version of PHP 5.6.0 and maximum version of PHP 8.1.*.\n");
```
Comment (mvorisek) [inline php-cs-fixer:27]:
```suggestion
    fwrite(STDERR, "PHP needs to be a minimum version of PHP 5.6.0 and maximum version of PHP 8.0.*.\n");
```

Comment (Nyholm) [reply php-cs-fixer:27]: oops. Thank you

**Comment by Nyholm** [comment]
I've tested this PR on my CI. It is working as excepted on PHP 8 =)

**Comment by MatTheCat** [comment]
Maybe an additional check is needed for PHP 8.0.0: because of https://bugs.php.net/bug.php?id=80462 `?->` tokens will be missing.

**Comment by Nyholm** [comment]
I dont agree. 
I think this if-statement should be removed completely. 

What we want is that people test this on PHP8 and report/fix bugs. Currently it is impossible to run the tool on PHP8 and the community is waiting impatiently on support and all the work falls on the maintainers. 

-------

I rather it works "most of the time", than I cannot use it at all. 

Also note: You can still say "we dont support PHP8" while allowing people to run the tool.

**Review by keradus — CHANGES_REQUESTED** [review]
.

**Comment by keradus** [comment]
Currently, the tool does not yet support named arguments, nullsafe operator, attributes and new union types.

The result running PHP CS Fixer on code with it is not known, currently.
We already had a case when PHP 8 code was parsed, changed by fixer and produced valid PHP syntax, but with different meaning. We don't want to offer that to users of the tool. In our opinion it's better to not offer the tool than offer the tool and break the code logic.

The mentioned case (That one is fixed already)
> if you run the tool on PHP8 and use the `NoUselessElseFixerTest` rule while not using the bootstrap file the following change will be done:
> 
> ```diff
> <?php
>              if ($foo) {
>                  $a = $condition or throw new Exception();
> -            } else {
> +            }  
>                  echo 123;
> -            }
> +            
> ```
> 
> It changes the code flow. 
> 
> Syntax is valid, linting of the file is all fine, everything passes on the tests, yet the code logic got changed.


Please, if you want to help with it, go to issue https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702 and help with one of the tasks. Thanks!
If not really, we need to ask for your patience. As you see, there are plenty of topics there, and we are on finish line with base support (we understand and don't break the code, the later step, not blocking the PHP v8 support, is to start having new rules for new syntax)



> I've tested this PR on my CI.

not sure what is your CI?

> You can still say "we dont support PHP8" while allowing people to run the tool.

so we do, use `PHP_CS_FIXER_IGNORE_ENV`

**Comment by Nyholm** [comment]
Thank you for a lengthy reply

> We don't want to offer that to users of the tool. In our opinion it's better to not offer the tool than offer the tool and break the code logic.

Im not sure I agree with this, but Im sure there are information that I haven't consider. Im happy that you considered this and made a decision.

> not sure what is your CI?

Sorry for being unclear. I ran my forked version of php-ci-fixer on my CI to test my projects. I have not run php-cs-fixer's testsuite. 

> so we do, use PHP_CS_FIXER_IGNORE_ENV

Thank you

**Pull request closed without merging**